File: frontend-angular/src/app/models/offer.ts
import {Shipping} from "./shipping";
import {User} from "./user";

export class Offer {

    id?: number;
    shippingCost?: number;
    message?: string;
    duration?: number;
    shippingId?: number;
    shipping?: Shipping
    distributorId?: number;
    distributor: User = new User();

    constructor() {
    }
}

File: backend-golang/app/models/offer.go
package models

import (
	"gorm.io/gorm"
)

// TODO: Field-Level Permission gorm
type Offer struct {
	gorm.Model
	ID            int `gorm:"primarykey"`
	ShippingCost  float64
	Message       string
	Duration      float64 // Estimated duration - hours
	ShippingID    int     // Has one relationship - Shipping
	DistributorID int
	Distributor   User      `gorm:"PRELOAD:true"`
	Shipping      *Shipping // Avoid recursive types
}
